always-hydrate.md

```diff
@@ -0,0 +1,130 @@
+is to not add any significant time to commands like `git checkout` that change the projection, so the
+implementation should ensure that these are done as background downloads that do not block these
+commands.
+
+## UX
+This is really an extension of the existing prefetch functionality. As such, it makes sense to tie
+the UX to the existing `gvfs prefetch` verb.
+
+Side note on the existing prefetch verb: it's really two different commands in one, and it's
+unfortunate that they are combined. The command `gvfs prefetch --commits` is entirely different in
+its goals, usage, and implementation than either of `gvfs prefetch --files` or `gvfs prefetch -
+folders` (and the latter two can be combined with each other, but not with `--commits`). Ideally we
+would have had `gvfs prefetch-commits` and `gvfs prefetch-files` as two separate commands, which
+would have then led to a cleaner design for this new addition. I'm open to other suggestions here
+if anyone sees a better pattern for extending the current verbs.
+
+###Proposed UX for always-hydrate commands:
+Adding new entries: 
+
+Add an `--always-hydrate` flag to the existing `--files` and `--folders` variants of `gvfs prefetch`.
+This will communicate that whatever list of folders, files, and wildcards that were passed in to the
+command should be saved and reapplied after every projection change
+
+Syntax:
+
+```
+    gvfs prefetch 
+      [<Repo Root Path>]
+      [--files <semicolon-delimited list of file paths, prefix * wildcards allowed>] 
+      [--folders <semicolon-delimited list of folder paths, no wildcards>]
+      [--folders-list <path of file containing newline-delimited paths of folders, no wildcards>] 
+      [--[always-]hydrate]
+```
+
+The only change from today's syntax is the optional `[always-]` on the hydrate flag. `--always
+hydrate` implies `--hydrate`.
+
+Passing in the `--always-hydrate` flag has all the same behaviors that `prefetch --hydrate` has today
+and also adds the specified files and folders to the `always-hydrate` file, so that those files will
+continue to be hydrated in the future.
+
+Managing existing entries:
+
+In addition to adding entries to the `always-hydrate` file, users will also want to list out the
+existing entries and remove some or all of the existing entries. To that end, we can support the
+following commands:
+
+```
+    gvfs prefetch
+      [<Repo Root Path>]
+      --list-always-hydrate
+```
+
+The `--list-always-hydrate` flag is incompatible with any of the other flags. It will read out the
+`always-hydrate` file and display the entries to the user.
+
+```
+    gvfs prefetch
+      [<Repo Root Path>]
+      [--files <semicolon-delimited list of file paths, prefix * wildcards allowed>] 
+      [--folders <semicolon-delimited list of folder paths, no wildcards>]
+      --remove-always-hydrate
+```
+
+Any entries that match the specified `--files` and `--folders` arguments will be removed from the
```

Comment:
Note: `--remove-always-hydrate` is incompatible with `--[always-]hydrate` or `--list-always-hydrate`